File: 3 Longest Substring Without Repeating Characters.c
int lengthOfLongestSubstring(char* s) {
    if (!s[0]) return 0;
	char you[256] = { 0 };
	int i = 1, p = 0, ans = 1;
	you[s[0]] = 1;
	for (; s[i]; i++) {
		while (you[s[i]]) {
			you[s[p++]] = 0;
		}
		you[s[i]] = 1;
		if (i - p + 1 > ans) ans = i - p + 1;
	}
	if (i - p > ans) ans = i - p;
	return ans;
}

File: 3 Longest Substring Without Repeating Characters.java
public class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.equals("")) return 0;
        boolean[] you = new boolean[256];
        int p = 0, ans = 1;
        you[s.charAt(0)] = true;
        for (int i = 1; i < s.length(); i++) {
            while (you[s.charAt(i)]) {
                you[s.charAt(p++)] = false;
            }
            you[s.charAt(i)] = true;
            ans = Math.max(ans, i - p + 1);
        }
        return ans;
    }
}
